packages/doenetml-worker-rust/lib-doenetml-core/src/core/dast/macro_expand.test.rs

```diff
@@ -0,0 +1,240 @@
+use assert_json_diff::assert_json_eq;
+
+use super::*;
+use crate::test_utils::*;
+
+#[test]
+fn macros_get_expanded_to_their_referents() {
+    let dast_root = dast_root_no_position(r#"<point name="p" />$p"#);
+    let mut flat_root = FlatRoot::from_dast(&dast_root);
+    Expander::expand(&mut flat_root);
+    assert_json_eq!(
+        serde_json::to_value(&flat_root).unwrap(),
+        json!(
+            {
+                "type": "FlatRoot",
+                "children": [0],
+                "nodes": [
+                  {
+                    "type": "Element",
+                    "name": "document",
+                    "children": [1, 2],
+                    "attributes": [],
+                    "idx": 0
+                  },
+                  {
+                    "type": "Element",
+                    "name": "point",
+                    "parent": 0,
+                    "children": [],
+                    "attributes": [
+                      {
+                        "name": "name",
+                        "parent": 1,
+                        "children": ["p"]
+                      }
+                    ],
+                    "idx": 1
+                  },
+                  {
+                    "type": "Element",
+                    "name": "point",
+                    "parent": 0,
+                    "children": [],
+                    "attributes": [],
+                    "idx": 2,
+                    "extending": {
+                      "node_idx": 1,
+                      "unresolved_path": null
+                    }
+                  }
+                ]
+              }
+        )
+    );
+
+    let dast_root = dast_root_no_position(
+        &r#"<a name="x">
+            <b name="y">
+                <c name="z" />
+                $x.y
+            </b>
+            <f>
+                <e name="w" />
+            </f>
+        </a>
+        $w
+        <d name="y" />"#
+            .replace("\n", "")
+            .replace("  ", ""),
```

Comment:
I'm curious why you put these replaces in here. Can't the `dast_root_no_position()` function handle white space?

Reply:
Oh, it's probably for the comparison to the `.to_xml()` output, below.

Reply:
It's easier to see the layout indented, but those extra text nodes clutter up the output, so I preemtively removed them.